python_ga/utils.py

```diff
@@ -148,16 +229,52 @@ def resolve_saving_path(path: str = 'experiment_results', config=None):
     return saving_directory
 
 
+# ==CONFIG FILE READER==================================================================================================
 def read_config(config_path: str):
+    """Odczyt konfiguracji z pliku Yaml."""
     with open(config_path, 'r') as file:
         config = yaml.load(file, yaml.FullLoader)
     return config
 
 
+# ==PROBABILITY ANNEALING===============================================================================================
 def calculate_probability_annealing(iteration, max_value=0.5, max_iter=100, eps=0.001, rounding_decimal=6):
+    """Symulowanie wyżarzanie."""
     def formula(x):
+        """Formuła do obliczenia prawdopodobieństw przy wyżarzaniu."""
         result = np.round((x ** 4) * max_value, rounding_decimal)
         return result if result >= eps else eps
 
     x = 1 - iteration / max_iter
     return formula(x)
+
+
+# ==FITNESS CALCULATION=================================================================================================
+def calculate_fitness(paired_population, model):
```

Comment:
moved to another file

works on `paired_population` instead of taking `population` as a list and converting them internally